ios/FluentUI/Vnext/IndeterminateProgressBar/IndeterminateProgressBar.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+import UIKit
+
+/// UIKit wrapper that exposes the SwiftUI Indeterminate Progress Bar implementation
+@objc open class MSFIndeterminateProgressBar: NSObject, FluentUIWindowProvider {
+
+    /// The UIView representing the Indeterminate Progress Bar.
+    @objc open var view: UIView {
+        return hostingController.view
+    }
+
+    /// The object that groups properties that allow control over the Indeterminate Progress Bar appearance.
+    @objc open var state: MSFIndeterminateProgressBarState {
+        return self.indeterminateProgressBarView.state
+    }
+
+    /// Creates a new MSFIndeterminateProgressBar instance.
+    @objc public override convenience init() {
+        self.init(theme: nil)
+    }
+
+    /// Creates a new MSFIndeterminateProgressBar instance.
+    /// - Parameters:
+    ///   - theme: The FluentUIStyle instance representing the theme to be overriden for this Indeterminate Progress Bar
+    @objc public init(theme: FluentUIStyle? = nil) {
+        super.init()
+
+        indeterminateProgressBarView = IndeterminateProgressBar()
+        hostingController = UIHostingController(rootView: AnyView(indeterminateProgressBarView
+                                                                    .windowProvider(self)
+                                                                    .modifyIf(theme != nil, { indeterminateProgressBarView in
+                                                                        indeterminateProgressBarView.customTheme(theme!)
+                                                                    })))
+        hostingController.disableSafeAreaInsets()
+        view.backgroundColor = UIColor.clear
+    }
+
+    var window: UIWindow? {
+        return self.view.window
+    }
+
+    private var hostingController: UIHostingController<AnyView>!
+
+    private var indeterminateProgressBarView: IndeterminateProgressBar!
+}
+
+/// Properties available to customize the state of the Indeterminate Progress Bar state
+@objc public protocol MSFIndeterminateProgressBarState {
+    /// Defines whether the Indeterminate Progress Bar is animating or stopped.
+    var isAnimating: Bool { get set }
+
+    /// Defines whether the Indeterminate Progress Bar is visible when its animation stops.
+    var hidesWhenStopped: Bool { get set }
+}
+
+/// View that represents the Indeterminate Progress Bar
+public struct IndeterminateProgressBar: View {
```

Comment:
I think there needs to be bit more documentation on why and what is different from swiftui progressive which does both determinate and indeterminate style. at some point, can we move towards the os api? can we use similar configuration struct and style protocols? https://developer.apple.com/documentation/swiftui/progressview

Reply:
The ProgressView is iOS 14+ only. This is pretty much the "why" part of it.
In terms of APIs, we're building "one-to-one" matching APIs to replace the legacy controls in the smoothest way possible in UIKit. In SwiftUI we are trying to make our API patterns consistent across controls (modifiers changing state objects). 

We also support some things that the ProgressView does not:
- Start/stopping the animation of the bar being one example.
- Keep the view visible when the animation is stopped
- Indeterminate style as a bar instead of just spinner.

I added a note about iOS 14+. Let me know your thoughts there.

Reply:
yup wanted to be super explicit in the documentation why. also are you deleted the fluent ui kit one now or later?

Reply:
The process we're doing is that we delete the UIKit version when we merge the Vnext control to the main branch.